src/OrchardCore.Modules/OrchardCore.Google/TagManager/Deployment/GoogleTagManagerDeploymentStepDriver.cs

```diff
@@ -0,0 +1,23 @@
+using OrchardCore.Deployment;
+using OrchardCore.DisplayManagement.Handlers;
+using OrchardCore.DisplayManagement.Views;
+
+namespace OrchardCore.Google.TagManager.Deployment
+{
+    public class GoogleTagManagerDeploymentStepDriver : DisplayDriver<DeploymentStep, GoogleTagManagerDeploymentStep>
+    {
+        public override IDisplayResult Display(GoogleTagManagerDeploymentStep step)
```

Comment:
We should be able to skip all the `DeploymentStep` and `RecipeStepHandler` for these by using the newer generic site settings methods

See here for when I am improving it (but the old code works just fine too)

https://github.com/OrchardCMS/OrchardCore/pull/9728

just saves having more unecesary code to maintain for deployments and recipes.

Other than that all looks good

Reply:
Can we merge and fix the remaining when your PR is merged. I have also changes in the Google Analytics feature to allow using multiple MeasurementId's that I will push in a second PR.

Reply:
The idea is to not have these classes views at all.
So for now you can still use the existing deployment / recipe registration (my pr is just some nicer registration, so I can update that later after this is merged)

```
            services.AddTransient<IDeploymentSource, SiteSettingsPropertyDeploymentSource< GoogleTagManagerSettings >>();
            services.AddScoped<IDisplayDriver<DeploymentStep>>(sp =>
            {
                var S = sp.GetService<IStringLocalizer<DeploymentStartup>>();
                return new SiteSettingsPropertyDeploymentStepDriver< GoogleTagManagerSettings >(S["Admin settings"], S["Exports the admin settings."]);
            });
            services.AddSingleton<IDeploymentStepFactory>(new SiteSettingsPropertyDeploymentStepFactory< GoogleTagManagerSettings >());
```

So if you add this instead (with the localization), to the`Startup`, and then you can delete all the deployment views / drivers and recipe steps - if we do it after merge, then we get stuck with a bunch of code that can't be removed.

Just about less code to maintain ;)

Reply:
ok done